implementations/micrometer-registry-datadog/src/main/java/io/micrometer/datadog/DatadogConfig.java

```diff
@@ -56,12 +58,56 @@ default String hostTag() {
         return getString(this, "hostTag").orElse("instance");
     }
 
+    /**
+     * Truthy if the configuration is defined for a dogstatsd reporting configuration.
+     * @return Returns true if this is a non https scheme target, to indicate to talk via
+     * dogstatsd.
+     */
+    default boolean isStatsd() {
+        String scheme = URI.create(uri()).getScheme();
+        return !(scheme.equalsIgnoreCase("https") || scheme.equalsIgnoreCase("http"));
+    }
+
+    /**
+     * Turn this on to enable client-side (in process) aggregation for dogstatsd mode.
+     * @return Returns true if aggregation is enabled.
+     */
+    default boolean enableAggregation() {
```

Comment:
Adding these methods to `DatadogConfig` is one approach that could work, but I'm concerned about the complexity for users of having some methods that only do anything when running in one mode but not the other. It's not unprecedented in Micrometer, but it's nice to keep things as simple as we can.

Do you have an idea how common a need it is for users to customize these aggregation configuration properties?

If the defaults are acceptable for most users, I think we could consider ways to allow customization without adding methods to `DatadogConfig`. One idea I mentioned in https://github.com/micrometer-metrics/micrometer/pull/3329#discussion_r942267577 was allowing users to provide their own `StatsDClient` to the `DatadogMeterRegistry.Builder` if they want to have the ability to fully customize. We could still allow users to get other config from the `DatadogConfig` with a method that returns a `NonBlockingStatsDClientBuilder` with e.g. the host set. Does that make sense? What do you think?